playbooks/settings.yml

```diff
@@ -18,6 +18,17 @@ misc:
     statedir: "/tmp"
     configdir: "/tmp/config"
 
+paths:
+  samba:
+    etc: /etc/samba
+
+  ctdb:
+    var_lib: /var/lib/ctdb
+    etc: /etc/ctdb
+    bin: /bin
+
+  mount: /mnt
```

Comment:
Apart from `mount` all other path definitions seems a bit overkill to me. I would let each usage address the path directly.

Reply:
The problem is that we use it in many places. If for some reason we decide to move them elsewhere, changing the hardcoded paths will be harder. On the other side, this approach makes it much easier for the user to customize paths for their own environment (for example, installation path in /usr/local instead of the normal paths if samba and/or ctdb are installed from sources for developing/debugging reasons). It could even be used to automatically collect configuration settings from different components by automatically copying the _etc_ path from each component.

Having them on variables makes it possible to create a patch to easily override them from the `EXTRA_VARS` variable or other methods by the user.

Reply:
Thinking about source installations we could use the `prefix` approach so that standardized paths can be optionally prefixed with a user defined location.

Having said that we can start with this basic approach and see if in future it calls for the other approach.